### ISSUE TITLE
Support Ruby 2.7 - fix depreciations, warnings, and add to CircleCI

### ISSUE THREAD

**Issue by joshdholtz** [fastlane/fastlane#16342]
Ruby 2.7 breaks a bunch of test. There is a decent list of things that need to get addressed. 

I will add related issues and PR pointed to this issue until everything is fixed.

## Related Issues
- https://github.com/fastlane/fastlane/issues/15856

**Comment by Kaspik** [comment]
I think this should be re-opened as Ruby 2.7 is definitely not fully-compatible.

**Comment by janpio** [comment]
Can you be more precise or point to individual issues @Kaspik?

**Comment by Kaspik** [comment]
Check all open PRs in last 3 days with Ruby 2.7 title :)

**Comment by janpio** [comment]
https://github.com/fastlane/fastlane/labels/topic%3A%20ruby%202.7 should do it now :)

**Comment by Kaspik** [comment]
I don't think we want to keep creating separate issue for warnings, so let's accumulate all warnings we find here? :) Or do you want separate issue with given tag?

```
/Users/kaspik/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/fastlane-2.148.1/fastlane_core/lib/fastlane_core/ui/interface.rb:141: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/Users/kaspik/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/fastlane-2.148.1/fastlane_core/lib/fastlane_core/ui/errors/fastlane_error.rb:9: warning: The called method `initialize' is defined here
```

**Comment by janpio** [comment]
Open PR that adds 2.7 to CI, currently has 10 failures:
https://github.com/fastlane/fastlane/pull/16407

**Comment by Kaspik** [comment]
Still not resolved.

**Comment by Kaspik** [comment]
Not sure but probably fixed?

**Comment by macabeus** [comment]
@Kaspik I just installed Fastlane now and I'm receiving many warnings, such as:
```
➜  android git:(master) ✗ bundle exec fastlane beta
[✔] 🚀 
[15:43:00]: ------------------------------
[15:43:00]: --- Step: default_platform ---
[15:43:00]: ------------------------------
/usr/local/Cellar/fastlane/2.154.0/libexec/gems/fastlane-2.154.0/fastlane_core/lib/fastlane_core/ui/interface.rb:141: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/usr/local/Cellar/fastlane/2.154.0/libexec/gems/fastlane-2.154.0/fastlane_core/lib/fastlane_core/ui/errors/fastlane_error.rb:9: warning: The called method `initialize' is defined here
```

Maybe still there are some warnings that need to be fixed?

**Comment by joshdholtz** [comment]
Hey all! I'll be getting back to this now 😊 My focus had to get changed to adapting the new App Store Connect API but that is all good now 🤞 Hopefully should have fixes/improvements for Ruby 2.7 soon!

**Comment by sezabass** [comment]
Still not fixed
